src/rust/inetstack/protocols/tcp/established/background/acknowledger.rs

```diff
@@ -33,16 +33,20 @@ pub async fn acknowledger<const N: usize>(mut cb: SharedControlBlock<N>) -> Resu
         futures::pin_mut!(ack_deadline_changed);
 
         let clock_ref: SharedTimer = cb.get_timer();
-        let yielder: Yielder = Yielder::new();
         let ack_future = match deadline {
-            Some(t) => Either::Left(clock_ref.wait_until(t, yielder).fuse()),
+            Some(t) => Either::Left(clock_ref.wait_until(t, &yielder).fuse()),
             None => Either::Right(future::pending()),
         };
         futures::pin_mut!(ack_future);
 
         futures::select_biased! {
             _ = ack_deadline_changed => continue,
             _ = ack_future => {
+                match cb.get_ack_deadline().get() {
+                    Some(timeout) if timeout > cb.get_now() => continue,
+                    None => continue,
+                    _ => {},
```

Comment:
Is there a possible `_` condition if we have already covered `Some` and `None`?

Reply:
Yes if timeout is less than cb.get_now(), then we actually want to send the ack.

Reply:
Oh yes, I missed the condition. This loosk good. Thanks.